File: win-prog/ProgWinVB/Paths Regions and Clipping/LineArcPath/LineArcPath.vb
'--------------------------------------------
' LineArcPath.vb (c) 2002 by Charles Petzold
'--------------------------------------------
Imports System
Imports System.Drawing
Imports System.Drawing.Drawing2D
Imports System.Windows.Forms

Class LineArcPath
    Inherits PrintableForm

    Shared Shadows Sub Main()
        Application.Run(New LineArcPath())
    End Sub

    Sub New()
        Text = "Line & Arc in Path"
    End Sub

    Protected Overrides Sub DoPage(ByVal grfx As Graphics, _
            ByVal clr As Color, ByVal cx As Integer, ByVal cy As Integer)
        Dim path As New GraphicsPath()
        Dim pn As New Pen(clr, 25)

        path.AddLine(25, 100, 125, 100)
        path.AddArc(125, 50, 100, 100, -180, 180)
        path.AddLine(225, 100, 325, 100)

        grfx.DrawPath(pn, path)
    End Sub
End Class

File: win-prog/ProgWinCS/Paths Regions and Clipping/LineArcPath/LineArcPath.cs
//------------------------------------------
// LineArcPath.cs © 2001 by Charles Petzold
//------------------------------------------
using System;
using System.Drawing;
using System.Drawing.Drawing2D;
using System.Windows.Forms;

class LineArcPath: PrintableForm
{
     public new static void Main()
     {
          Application.Run(new LineArcPath());
     }
     public LineArcPath()
     {
          Text = "Line & Arc in Path";
     }
     protected override void DoPage(Graphics grfx, Color clr, int cx, int cy)
     {
          GraphicsPath path = new GraphicsPath();
          Pen          pen  = new Pen(clr, 25);

          path.AddLine( 25, 100, 125, 100);
          path.AddArc (125,  50, 100, 100, -180, 180);
          path.AddLine(225, 100, 325, 100);

          grfx.DrawPath(pen, path);
     }
}
